tests/manage/mcg/test_credentials_reset.py

```diff
@@ -0,0 +1,234 @@
+            "password"
+        ]
+
+        def finalizer():
+            """
+            Reset the password back to the original and retrieve a new RPC token
+
+            """
+            mcg_obj_session.reset_admin_pw(new_password=original_password)
+            mcg_obj_session.retrieve_nb_token()
+
+        request.addfinalizer(finalizer)
+        return original_password
+
+    @tier2
+    @polarion_id("OCS-5118")
+    def test_change_nb_admin_pw(self, mcg_obj_session, original_noobaa_admin_password):
+        """
+        Test changing the NooBaa admin password
+
+        1. Change the noobaa-admin password
+        2. Verify the password changed in the noobaa-admin secret
+        3. Verify the old password fails when attempting to generate an RPC token and the new password succeeds
+
+        """
+
+        # Change the noobaa-admin password
+        new_password = f"new_nb_admin_password-{(uuid.uuid4().hex)[:16]}"
+        mcg_obj_session.reset_admin_pw(new_password=new_password)
+
+        # Verify the password changed in the noobaa-admin secret
+        assert (
+            new_password
+            == mcg_obj_session.get_noobaa_admin_credentials_from_secret()["password"]
+        )
+
+        # Verify the original password fails when attempting to generate an RPC token
+        mcg_obj_session.noobaa_password = original_noobaa_admin_password
+        try:
+            with pytest.raises(AssertionError):
+                mcg_obj_session.retrieve_nb_token(timeout=30, sleep=10)
+                logger.error(
+                    "Unexpectedly succeeded in retrieving RPC token with the original password"
+                )
+        except Exception as e:
+            logger.error(f"An unexpected exception occurred: {e}")
+            # Set the new password to allow resetting back at teardown
+            mcg_obj_session.noobaa_password = new_password
+            raise e
+
+        # Verify the new password succeeds when attempting to generate an RPC token
+        mcg_obj_session.noobaa_password = new_password
+        try:
+            mcg_obj_session.retrieve_nb_token(timeout=60)
+            logger.info("Successfully retrieved RPC token with new password")
+        except Exception as e:
+            logger.error(f"Failed to retrieve RPC token with new password: {e}")
+            raise
+
+    @tier2
+    @polarion_id("OCS-5119")
+    def test_regenerate_account_s3_creds(
+        self, mcg_obj_session, mcg_account_factory, bucket_factory
+    ):
+        """
+        Test regenerating S3 credentials for an MCG account
+
+        1. Create a new noobaa account and fetch its S3 credentials
+        2. Regenerate its S3 credentials using the noobaa CLI command
+        3. Verify its S3 credentials have changed at the secret
+        4. Verify that creating an S3 bucket fails with the old credentials and succeeds with the new credentials
+
+        """
+        acc_name = f"credentials-reset-acc-{random.randrange(100)}"
```

Comment:
@sagihirshfeld are you trying to set a name here for the default account?

Reply:
Nope, this name is for a new noobaa account that we later reset its S3 credentials by using `mcg-cli account regenerate {acc_name}`.

While it is possible to reset the S3 credentials of the default noobaa-admin account, it could affect other tests so I preferred not to.